Source/Helpers/Log.swift

```diff
@@ -42,3 +42,88 @@ public func Log (_ message: String,
     print("(\(queue)) \(fileURL) \(function) [\(line)]: " + message)
   #endif
 }
+
+#if DEBUG
+/// Helper log method for more accurate console prints.
+///
+/// Prints using OSLog. The `logAccess` set the visibility of the logs.
+/// If it is private the log message doesn't appear in the logs console when
+/// the device isn't connected.
+///
+/// https://developer.apple.com/documentation/os/logging
+///
+/// - Parameters:
+///   - message: Object to be printed. Can be an expression.
+///   - log: OSLog object for this log.
+///   - type: OSLogType for this log.
+///   - logAccess: the level of privacity of the log. By default is public.
+///   - function: Name of the function calling this function. By default, same name as function where this is called.
+///   - line: Line number where this function is called. By default, line number within the file where the call is made.
+@available(OSX 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
+public func Log(_ message: String,
+                log: OSLog,
+                type: OSLogType = OSLogType.default,
+                _ logAccess: LogAccess = .public,
+                _ function: StaticString = #function,
+                _ line: Int = #line) {
+  
+  let customMessage = "\(type.descriptionLog) \(function) [\(line)]: \(message)"
+  os_log(logAccess.toStaticString(), log: log, type: type, customMessage)
+}
+
+#else
+
+/// Helper log method for more accurate console prints.
+///
+/// Prints using OSLog. The `logAccess` set the visibility of the logs.
+/// If it is private the log message doesn't appear in the logs console when
+/// the device isn't connected.
+///
+/// https://developer.apple.com/documentation/os/logging
+///
+/// - Parameters:
+///   - message: Object to be printed. Can be an expression.
+///   - log: OSLog object for this log.
+///   - type: OSLogType for this log.
+///   - logAccess: the level of privacity of the log. By default is public.
+@available(OSX 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
+public func Log(_ message: String,
+                log: OSLog,
+                type: OSLogType = OSLogType.default,
+                _ logAccess: LogAccess = .public) {
+  os_log(logAccess.toStaticString(), log: log, type: type, message)
+}
+#endif
+
+@available(OSX 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
+extension OSLogType {
+  
+  fileprivate var descriptionLog: String {
+    switch self {
+    case OSLogType.info:
+      return "ℹ️ℹ️ℹ️(info)"
+    case OSLogType.debug:
+      return "🔹🔹🔹(debug)"
+    case OSLogType.error:
+      return "‼️‼️‼️(error)"
+    case OSLogType.fault:
+      return "💣💣💣(fault)"
+    default:
+      return "⬜️⬜️⬜️(default)"
```

Comment:
I see three grey squares. Is that correct?

Reply:
Yes

Reply:
Ok